Support plotting for entities with / in the name

scpeters opened the pull request:
I noticed while testing the rexrov models from
Field-Robotics-Lab/dave that the plotting window
does not work properly when elements of a model include
a forward-slash `/` in the entity name. To fix the problem,
the `/` characters are encoded as %2f in physics::Base::URI().
Corresponding decoding is added to the gui/plot/Palette class.

Fixes #3181.

scpeters committed:
Support plotting for entities with / in the name

I noticed while testing the rexrov models from the rexrov
model in Field-Robotics-Lab/dave that the plotting window
does not work properly when elements of a model include
a forward-slash `/` in the entity name. To fix the problem,
the `/` characters are encoded as %2f in physics::Base::URI().
Corresponding decoding is added to the gui/plot/Palette class.

Signed-off-by: Steve Peters <scpeters@openrobotics.org>

scpeters committed:
Try string::compare instead of == to fix build

I noticed a CI build failure with gcc-8 that doesn't occur
with gcc-9. Try to avoid failure by using std::string's
compare function instead of an == operator with a
const char * string literal.

Signed-off-by: Steve Peters <scpeters@openrobotics.org>

iche033 commented on gazebo/physics/Base.cc:
```diff
@@ -383,15 +384,19 @@ common::URI Base::URI() const
     {
-      uri.Path().PushFront(p->GetName());
+      std::string escapedParentName = p->GetName();
+      boost::replace_all(escapedParentName, "/", "%2f");
```
use `common::replaceAll` instead of `boost`? I do still see `boost` being used in gazebo so maybe not a big deal

scpeters replied:
I switched to `common::replaceAll` in https://github.com/osrf/gazebo/pull/3187/commits/7edb013c5c86738ea6489609f042b099f9707c67

iche033 reviewed (approved):
looks good, test passes for me. Just one minor comment but PR is fine as is.

scpeters committed:
Use common::replaceAll instead of boost

Signed-off-by: Steve Peters <scpeters@openrobotics.org>

scpeters committed:
Escape '%' as "%25" as well

This would prevent a string containing "%2f" from
incorrectly being unescaped to "/".

Signed-off-by: Steve Peters <scpeters@openrobotics.org>
